### PR TITLE
Renaming HostRequest to_user --> host and from_user --> surfer

### PR DESCRIPTION
**Pull request by ShmuelTreiger**
This got me all confused when reviewing Aapeli's pull request, so I went ahead and refactored it (as briefly mentioned in the slack channel)

This will need frontend work as well

Also, unsure if the migrations will work as the test is broken on my machine, so well see

**Backend checklist**
- [x] Formatted my code by running `autoflake -r -i --remove-all-unused-imports src && isort . && black .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [x] All tests pass
- [ ] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Frontend checklist**
- [ ] Formatted my code with `yarn format && yarn lint --fix`
- [ ] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
Could we merge https://github.com/Couchers-org/couchers/pull/1623 first?

**Comment by ShmuelTreiger** [comment]
No problem. It will need frontend changes anyway.

**Comment by aapeliv** [comment]
Btw, there's no need to change the tag number in protos if the meaning and type stay the same and you're just changing the name

**Comment by ShmuelTreiger** [comment]
Had no idea, thanks!

**Comment by ShmuelTreiger** [comment]
Done/ready for front end post #1623

**Review by kalvdans — DISMISSED** [review]
Very nice clarification of the roles. redo the migrations after #1623 has been merged.

File: app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_constraint("fk_host_requests_from_user_id_users", "host_requests", type_="foreignkey")
+    op.drop_constraint("fk_host_requests_to_user_id_users", "host_requests", type_="foreignkey")
+    op.create_foreign_key(op.f("fk_host_requests_host_id_users"), "host_requests", "users", ["host_id"], ["id"])
+    op.create_foreign_key(op.f("fk_host_requests_surfer_id_users"), "host_requests", "users", ["surfer_id"], ["id"])
+    op.drop_index("ix_host_requests_from_user_id", table_name="host_requests")
+    op.drop_index("ix_host_requests_to_user_id", table_name="host_requests")
+    op.create_index(op.f("ix_host_requests_host_id"), "host_requests", ["host_id"], unique=False)
+    op.create_index(op.f("ix_host_requests_surfer_id"), "host_requests", ["surfer_id"], unique=False)
```
Comment (aapeliv) [inline app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: Are you not able to just rename the indexes/constraints? This is fine-ish, but creating constraints/indexes can take quite a long time on larger databases and might take a whole table write lock

Comment (ShmuelTreiger) [reply app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: It looks like no and that its an sql thing, not an sqla thing:

https://stackoverflow.com/questions/44631175/alembic-how-to-alter-a-foreigenkey-column 
https://stackoverflow.com/questions/13606469/cannot-change-column-used-in-a-foreign-key-constraint

And yes it requires a table lock (didn't even think of this). Will sqla do that automatically? Or do I need to add it?

**Comment by ShmuelTreiger** [comment]
Ready for review and then frontend implementation

File: app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py (hunk 1 above)
Comment (aapeliv) [reply app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: You can rename them in postgres:

https://www.postgresql.org/docs/current/sql-alterindex.html

https://www.postgresql.org/docs/current/sql-altertable.html

File: app/proto/requests.proto [hunk 2]
```diff
@@ -40,7 +40,7 @@ service Requests {
 }
 
 message CreateHostRequestReq {
-  int64 to_user_id = 1;
+  int64 host_id =1;
```
Comment (aapeliv) [inline app/proto/requests.proto]: Space

**Review by aapeliv — COMMENTED** [review]
Did you fix the new instances?

**Comment by ShmuelTreiger** [comment]
From #1623? Yes

File: app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py (hunk 1 above)
Comment (ShmuelTreiger) [reply app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: I renamed them just now without thinking. It's not about renaming them, it's about applying the same constraint to a newly named column. I don't see a way to do that?

Comment (aapeliv) [reply app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: What do you mean? The column didn't change, just the name. The constraint still applies to that renamed column

Comment (ShmuelTreiger) [reply app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: Will the index/fk just track the column post-rename?

Comment (aapeliv) [reply app/backend/src/couchers/migrations/versions/d1a68f873ee3_rename_hostrequest_user_columns.py]: Yep. The names are just labels, the data/indexes/constraints are stored separately

**Comment by aapeliv** [comment]
This seems fine, just needs frontend work

**Review by aapeliv — CHANGES_REQUESTED** [review]
Actually, it seems to me that our convention has been to use `host_user_id` (not `host_id`), so we should stick to that. (See `Page` model, `Complaint`, etc. There's some cases where it's not followed though.)

**Comment by aapeliv** [comment]
I can then do the frontend bit

**Comment by ShmuelTreiger** [comment]
I'm not sure why gitlab isn't running? But I updated the names @aapeliv

**Review by aapeliv — COMMENTED** [review]
Backend LGTM

**Pull request merged**